ledgerManager.php

```diff
@@ -0,0 +1,319 @@
+			delete from {$this->table}
+			where id = :transactionId
+		");
+		return $query->execute([':transactionId' => $transactionId]);
+	}
+
+	/**
+	 * This method returns the sum of all credit or debit transactions
+	 * for (an optional) date range
+	 *
+	 * @param bool $credit
+	 * @param string|null $startDate
+	 * @param string|null $endDate
+	 * @return float
+	 */
+	protected function getTotalSumCreditTransactions($credit = true, $startDate = null, $endDate = null)
+	{
+		$queryString = "
+			select sum(amount) amount
+			from {$this->table}
+			where credit = :credit
+		";
+		if ($startDate) {
+			$queryString .= "and time >= :startDate";
+		}
+		if ($endDate) {
+			$queryString .= "and time <= :endDate";
+		}
+		$query = $this->pdo->prepare($queryString);
+		$query->bindParam(':credit', $credit);
+		if ($startDate) {
+			$query->bindParam(':startDate', $startDate);
+		}
+		if ($endDate) {
+			$query->bindParam(':endDate', $endDate);
+		}
+		$query->execute();
+		$row = $query->fetch(PDO::FETCH_ASSOC);
+		return (float)$row['amount'];
+	}
+
+	/**
+	 * This method returns a balance for a given date range, OR sets the value
+	 * of the current balance to the class property $overallBalance
+	 *
+	 * @param string|null $startDate
+	 * @param string|null $endDate
+	 * @return void|float
+	 */
+	protected function calculateBalance($startDate = null, $endDate = null)
+	{
+		$totalCredits = $this->getTotalSumCreditTransactions(true, $startDate, $endDate);
+		$totalDebits = $this->getTotalSumCreditTransactions(false, $startDate, $endDate);
+		$balance = $totalCredits - $totalDebits;
+		if (!$startDate && !$endDate) {
+			$this->overallBalance = $balance;
+		}
+		return $balance;
+	}
+
+	/**
+	 * @param string $startDate
+	 * @param string $endDate
+	 */
+	public function getCreditSumForWindow($startDate, $endDate)
+	{
+		return $this->getTotalSumCreditTransactions(true, $startDate, $endDate);
+	}
+
+	/**
+	 * @param string $startDate
+	 * @param string $endDate
+	 */
+	public function getDebitSumForWindow($startDate, $endDate)
+	{
+		return $this->getTotalSumCreditTransactions(false, $startDate, $endDate);
+	}
+
+	/**
+	 * This method returns a balance for a given date, now() by default
+	 *
+	 * @param string|null $startDate
+	 * @param string|null $endDate
+	 * @return float
+	 */
+	public function getBalance($startDate = null, $endDate = null)
+	{
+		if ($startDate || $endDate) {
+			return $this->calculateBalance($startDate, $endDate);
+		} elseif (!isset($this->overallBalance)) {
+			return $this->calculateBalance();
+		}
+		return $this->overallBalance;
+	}
+
+	/**
+	 * @return float
+	 */
+	public function getUnclearedAmount()
+	{
+		$query = $this->pdo->query("
+			select sum(amount) amount
+			from {$this->table}
+			where credit = false and cleared = false
+		");
+		$row = $query->fetch(PDO::FETCH_ASSOC);
+		return (float)$row['amount'];
+	}
+
+	/**
+	 * @return int
+	 */
+	public function numberOfDaysLeftInPayPeriod()
```

Comment:
this should be explained.  this assumes that one get paid on the 15th and last day of the of month (with no regard to weekends, b/c they don't matter in terms of how far an allowance should go).